tests/test_boot.py

```diff
@@ -10,8 +10,15 @@
 
 
 @pytest.mark.integration
-@pytest.mark.parametrize("format", [f for f in OutputFormat if f != OutputFormat.none])
-def test_boot(format: OutputFormat) -> None:
+@pytest.mark.parametrize(
+    "format",
+    [pytest.param(f, marks=getattr(pytest.mark, f"output_{f}")) for f in OutputFormat if f != OutputFormat.none],
+)
+@pytest.mark.parametrize(
+    "distribution",
+    [pytest.param(d, marks=getattr(pytest.mark, f"distro_{d}")) for d in Distribution if d != Distribution.custom],
```

Comment:
This gives me:
```python
[ParameterSet(values=(<Distribution.fedora: 'fedora'>,), marks=(MarkDecorator(mark=Mark(name='distro_fedora', args=(), kwargs={})),), id=None),
 ParameterSet(values=(<Distribution.debian: 'debian'>,), marks=(MarkDecorator(mark=Mark(name='distro_debian', args=(), kwargs={})),), id=None),
 ParameterSet(values=(<Distribution.ubuntu: 'ubuntu'>,), marks=(MarkDecorator(mark=Mark(name='distro_ubuntu', args=(), kwargs={})),), id=None),
 ParameterSet(values=(<Distribution.arch: 'arch'>,), marks=(MarkDecorator(mark=Mark(name='distro_arch', args=(), kwargs={})),), id=None),
 ParameterSet(values=(<Distribution.opensuse: 'opensuse'>,), marks=(MarkDecorator(mark=Mark(name='distro_opensuse', args=(), kwargs={})),), id=None),
 ParameterSet(values=(<Distribution.mageia: 'mageia'>,), marks=(MarkDecorator(mark=Mark(name='distro_mageia', args=(), kwargs={})),), id=None),
 ParameterSet(values=(<Distribution.centos: 'centos'>,), marks=(MarkDecorator(mark=Mark(name='distro_centos', args=(), kwargs={})),), id=None),
 ParameterSet(values=(<Distribution.rhel: 'rhel'>,), marks=(MarkDecorator(mark=Mark(name='distro_rhel', args=(), kwargs={})),), id=None),
 ParameterSet(values=(<Distribution.rhel_ubi: 'rhel-ubi'>,), marks=(MarkDecorator(mark=Mark(name='distro_rhel-ubi', args=(), kwargs={})),), id=None),
 ParameterSet(values=(<Distribution.openmandriva: 'openmandriva'>,), marks=(MarkDecorator(mark=Mark(name='distro_openmandriva', args=(), kwargs={})),), id=None),
 ParameterSet(values=(<Distribution.rocky: 'rocky'>,), marks=(MarkDecorator(mark=Mark(name='distro_rocky', args=(), kwargs={})),), id=None),
 ParameterSet(values=(<Distribution.alma: 'alma'>,), marks=(MarkDecorator(mark=Mark(name='distro_alma', args=(), kwargs={})),), id=None),
 ParameterSet(values=(<Distribution.gentoo: 'gentoo'>,), marks=(MarkDecorator(mark=Mark(name='distro_gentoo', args=(), kwargs={})),), id=None)]
```
I don't want to know. If it works, that must be good enough ;)